### ISSUE TITLE
Synchronization between instances of "BPN Discovery" at different operating companies

### ISSUE THREAD

**Issue by thomas-henn** [eclipse-tractusx/sig-release#35]
## Description
The "BPN Discovery" as a central service at each operating company in Catena-X is currently only able to orchestrate the discovery information per each operating environment. This will hinder the network-of-network approach. To enable the network-of-network approach with "BPN Discovery" services hosted centrally at each operating company, the databases behind the "BPN Discovery" services need to be synced.
A distributed database can enable this requirement.

## Impact
- Migration from local to distributed database in "BPN Discovery" service
- Authentication and auhtorization to distributed database needs to be aligned with self-sovereign identity access management
- Mixed operation with local and distributed database in "BPN Discovery" service possible

- [ ] I'm willing to contribute to this feature

### 1. Decentralization of Digital Twin Registry
![Figure 1: Decentralization of Digital Twin Registry](https://github.com/eclipse-tractusx/sig-release/files/12173376/Bild1.pdf)

![Figure 2: Central gap for discovery on Digital Twin Registries](https://github.com/eclipse-tractusx/sig-release/files/12173450/Bild2.pdf)

For the matter of data sovereignity the Digital Twin Registry (DTR) is moved from operation at an operating company to operation at data providers in the Catena-X network. Where the Fig. 1 shows the central approach, the Fig. 2 shows the decentral approach, where a DTR is hosted / operated at the site of a data provider.
Both figures show the data provider on the left side, the data consumer on the right side, and the network operator in between both. Both, the data provider and consumer host / operate an instance of the *Eclipse Dataspace Components* (EDC) for data share. Both instances of the EDC are under control of enterprise backend applications. Where the backend application(s) on the data consumer site triggers the data consume on the data provider site via both EDCs face-to-face, the backend application(s) on the data provider site supports the data consumer with requested data.

#### Overview
![Figure 3: Discovery services for Digital Twin Registries](https://github.com/eclipse-tractusx/sig-release/files/12173458/Bild3.pdf)

In the decentral approach of the DTR (see Figs. 2 and 3), the discovery of data assets and therefore Digital Twins needs to be ensured to avoid calls to every decentral DTR in an operating network of Catena-X. This is met by the use of three dedicated central services (at an operating company). This is the
1. Discovery Finder
2. BPN Discovery
3. EDC Discovery

The *Discovery Finder* is hosted under a fix address in an operating network and informs the data cosnumers and providers about the hosting endpoints of the BPN Discoveries and EDC Discovery, which can be in general dynamically hosted.

#### Proposal
The data discovery is proposed as the following:
1. POST-call to the *Discovery Finder* to find endpoints from *BPN Discoveries*
2. POST-call to dedicated *BPN Discovery* with input on *e.g.*, *specificAssetId* to find Legal Business Partner Number (BPN-L) of data provider
3. Call to *EDC Discovery* with input Legal Business Partner Number (BPN-L) to find Eclipse Dataspace Components (EDC) endpoint
4. Call to EDC endpoint to find Digital Twin Registries at data provider site
5. Call for discovery to Digital Twin Registry with respect to *specificAssetID* to find Digital Twin(s) and their aspect(s)
6. Call to data endpoint(s) from aspect(s) in Digital Twin(s)

#### API-endpoints on Discovery Finder
The API endpoints for the *Discovery Finder* are documented [here](https://semantics.int.demo.catena-x.net/discoveryfinder/swagger-ui/index.html "Open-API Specification of Discovery Finder")

In short, these describe the following:
1. For operating company
>- POST: Define endpoints to *BPN /EDC  Discovery*
>- DELETE: Delete endpoints to *BPN / EDC Discovery*
2. For data consumer
>- POST: Find endpoints from *BPN / EDC Discovery*

#### API-endpoints on BPN Discovery
The API endpoints for the *BPN Discovery* are documented [here](https://semantics.int.demo.catena-x.net/bpndiscovery/swagger-ui/index.html "Open-API Specification of BPN Discovery")

In short, these describe the following:
1. For data provider
>- POST: Create *number*-BPN matching(s)
>- DELETE: Delete *number*-BPN matching
2. For data consumer
>- POST: Find *number*-BPN matching(s)
